web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -0,0 +1,204 @@
+    }
+    // this is if you want to close the search bar with the search input focused
+    if (e.key === 'Escape') {
+      if (show) {
+        onEscape();
+        e.preventDefault();
+      }
+    }
+  };
+
+  const searchNext = () => {
+    search(searchInputRef.current.value, 'next');
+  };
+
+  const searchPrevious = () => {
+    search(searchInputRef.current.value, 'previous');
+  };
+
+  const onEscape = useCallback(() => {
+    onClose();
+    xTerm.getSearchAddon().clearDecorations();
+    xTerm.focus();
+  }, [onClose, xTerm]);
+
+  useEffect(() => {
+    const { unregister } = xTerm.registerCustomKeyEventHandler(e => {
+      if (e.type !== 'keydown') {
+        // continue through to xterm
+        return true;
+      }
+      if (isSearchKeyboardEvent(e)) {
+        onOpen();
+        searchInputRef.current?.focus();
+        e.preventDefault();
+        // event was handled and does not need to be handled by xterm
+        return false;
+      }
+      // this escape is handled if pressing escape with the term focused
+      if (e.key === 'Escape') {
+        onEscape();
+        e.preventDefault();
+        // event was handled and does not need to be handled by xterm
+        return false;
+      }
+
+      // continue through to xterm
+      return true;
+    });
+
+    return unregister;
+  }, [onEscape, onOpen, searchInputRef, xTerm, isSearchKeyboardEvent]);
+
+  if (!show) {
+    return;
+  }
+
+  const hasResults = searchResults.resultCount > 0;
+
+  return (
+    <SearchInputContainer alignItems="center" gap={3}>
+      <Input
+        ref={searchInputRef}
+        data-testid="terminal-search"
+        size="small"
+        autoFocus
+        onFocus={e => e.target.select()}
+        value={searchValue}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        width={150}
+      />
+      <Box minWidth={45}>
+        <P2>{`${searchResults.resultCount === 0 ? 0 : searchResults.resultIndex + 1}/${searchResults.resultCount}`}</P2>
+      </Box>
+      <Divider />
+      <Flex gap={1}>
+        <SearchButtonIcon
+          disabled={!hasResults}
+          title="Search previous"
+          onClick={searchPrevious}
+        >
+          <ChevronUp size="medium" />
+        </SearchButtonIcon>
+        <SearchButtonIcon
+          disabled={!hasResults}
+          title="Search next"
+          onClick={searchNext}
+        >
+          <ChevronDown size="medium" />
+        </SearchButtonIcon>
+        <SearchButtonIcon title="Close search" onClick={onEscape}>
+          <Cross size="medium" />
+        </SearchButtonIcon>
+      </Flex>
+    </SearchInputContainer>
+  );
+};
+
+const SearchInputContainer = styled(Flex)`
+  width: 365px;
+  padding: ${p => p.theme.space[2]}px;
+  padding-left: ${p => p.theme.space[3]}px;
+  padding-right: ${p => p.theme.space[3]}px;
+  background-color: ${p => p.theme.colors.levels.surface};
+  box-shadow: 0 8px 32px rgba(0, 0, 0, 0.1);
+  box-sizing: border-box;
```

Comment:
Same for this.

Reply:
im not sure what "same" you are referring to here.

Reply:
Ah sorry, I meant that we can remove it.
https://github.com/gravitational/teleport/pull/48776#discussion_r1846726178

Reply:
Ah ok. I removed `box-sizing`. The other props are like box shadow and background color are needed so I'll keep those